### ISSUE TITLE
clang-tidy false positive for performance-unnecessary-value-param

### ISSUE THREAD

**Issue by kees-jan** [llvm/llvm-project#62276]
Consider the following minimal example:
``` c++
#include <string>

struct Context
{
    static void set(std::string name, int value);
};

template <typename T>
void set(std::string name, T value)
{
    Context::set(std::move(name), value);
}

void f()
{
    set("test", 5);
}
```
clang-tidy 17 complains:
```
[<source>:9:22: warning: the parameter 'name' is copied for each invocation but only used as a const reference; consider making it a const reference [performance-unnecessary-value-param]](javascript:;)
void set(std::string name, T value)
                     ^
1 warning generated.
```
I believe this to be a false positive. The disassembly shows that the move constructor is being used (as it should be), not the copy constructor - no copy is made.

[See this example in compiler explorer](https://godbolt.org/z/r1vPP6PqY)

**Comment by firewave** [comment]
Related/duplicate: #99384
